lib/segment/src/index/field_index/geo_hash.rs

```diff
@@ -73,11 +74,11 @@ fn sphere_neighbor(hash_str: &str, direction: Direction) -> Result<GeoHash, Geoh
     let lat = sphere_lat(coord.y + 2f64 * lat_err.abs() * dlat);
 
     let neighbor_coord = Coord { x: lon, y: lat };
-    encode(neighbor_coord, hash_str.len())
+    encode(neighbor_coord, hash_str.len()).map(Into::into)
```

Comment:
I have trouble seeing how the Smol strings optimization helps.

It seems we still building heap allocated Strings and then turn them into a Smol string.

What am I missing?

Reply:
@timvisee your explanation is correct. `encode` method is external crate method, so I cannot change the signature of `encode`. But anyway only stack-based `SmolStr` will stay alive and heap-allocated `String` data will be dropped

Reply:
Thanks, makes sense! 
I thought we had some magic avoiding the first heap allocation completely :+1:

Reply:
Sad but no